src/Utils/AuthorizeFor.tsx

```diff
@@ -26,3 +35,16 @@ export const Anyone = () => true;
 export default function (userTypes: UserRole[]) {
   return (userType: UserRole) => userTypes.includes(userType);
 }
+
+export const AuthorizeUserRoute: React.FC<AuthorizeUserRouteProps> = ({
+  userTypes,
+  children,
+}) => {
+  const userData = useAuthUser();
+
+  if (userData && userTypes.includes(userData.user_type)) {
```

Comment:
the truthy check for `userData` is not required as the `useAuthUser` would always return a `UserModel` object and would always evaluate to true.

```suggestion
  const authUser = useAuthUser();
  if (userTypes.includes(authUser.user_type)) {
```